### ISSUE TITLE
Login: Bugs in access to topic via email invite

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#739]
User doesn't access the topic she should access after registering via join link, and there is a pop-up that appears twice. 

Here's how it goes: 

- She clicks 'join topic' link in email 

- She accepts this pop-up:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/93a1c012-1129-4e81-93c3-f9a4579127a4)

- She gets taken to the Login (not the registration) page, which does not have any social login or eID login options on it:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/9bf30715-b7d2-4e6c-985b-54700adc3267)

- She clicks 'Back to dashboard', clicks 'Login' on the homepage and it still has no social login options when she comes back to it: 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/63375530-3afc-480f-bcf0-b3551c47377d)

- She goes back to email invite to click Join topic again to try again, this time it’s different and she gets the register screen:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/b198a60d-68c3-4b7e-9e23-60e55efb063b)

- She registers
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/358e9300-c4dc-41bc-9bc0-8b9293f27b8e)

- She accepts this pop-up:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/a65d2da9-2e9a-4220-9003-03c1fcc5ec40)

- She gets taken here and logs in
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/2542a3cd-86e8-4b93-9ac9-78f47411e04b)

- She gets this pop-up again (now with incorrect info in yellow - she is already logged in so not relevant) that she has already agreed to
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/cf652519-e99e-473e-b2c8-31d55a10a6ed)

- She gets a terms of use pop-up even though she has already ticked it on sign up, and in the background the logged out dashboard appears:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/18171702-bcf4-4df0-bae6-36997c0e0d59)

- She continues without connecting ID
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/dc640561-682b-478d-aebc-e79dcd75e1b9)

- She skips the tour and in the background her logged in dashboard appears:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/383e0d62-d423-4e0d-8574-b6d8bf7a69bb)

- She now lands on her dashboard and can't see the topic, but she does find it in the activity feed so we carry on :) 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/a3575fc1-8021-4e9d-8c12-d95fa61a3974)

This is a bug issue for a few bugs - some development bugs some UX bugs: 
1. She should have been taken to the Register page
2. The login page that she was taken to by mistake doesn't have social logins when it should? 
3. Shouldn't the registration page also have social login options? 
4. She shouldn't get a pop-up to accept the T&Cs if she has already done so
5. She should land in the topic she was invited to, not on her dashboard 
6. She shouldn't have to click the 'Join topic' pop-up twice, only once 

But I will also make a UX insights issue where we can think about the login process, as wow what a lengthy process of pop-ups to navigate before finally accessing a topic! (Even if it had worked smoothly, it seems like far too many steps and pop-ups).

**Comment by ssin1901** [comment]
@BeccaMelhuish @ilmartyrk linked to #717 and quite frustrating for the user. I am adding the high priority label.

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @ssin1901 posted a fix. There was a flaw in the registration process where the user accepting terms was not properly handled. Also updated the registration flow to also automatically verify the users e-mail if they create a new account after clicking on the join link from the e-mail. The flow should work a lot better now. So for users who were invited and don't have an account, they will get the initial dialog with the create account button, then they will be taken to the registration flow. After filling and submitting they will be redirected to the invite dialog again and they will have login button there. After logging in they will be directly taken to the topic without any other dialogs

**Comment by BeccaMelhuish** [comment]
Thank you @ilmartyrk! Though @ssin1901 I think this relates to the issue that came up for your participant test this week? 

The use case was a public consultation, where the join link for the group of consultation topics was shared on social media. But as the user was new (as would very often be the case), after following the join link he had to register and then verify his email, but then when he verified his email he was taken to the dashboard, and couldn't find the group. @ssin1901 please correct or add anything if necessary :) 

So @ilmartyrk, should the fix you explained above work for join links that are just shared as a URL, as well as join links from a join email? Or if not, is there a solution to make the email verification button somehow send them to the group or topic they were initially trying to access, rather than sending them to the dashboard? 

EDIT: There's a [new issue](https://github.com/citizenos/citizenos-fe/issues/967) for this now, so this issue can be closed as long as all is working as it should (haven't tested it myself).
